src/lib/ngsi/EntityIdVector.cpp

```diff
@@ -120,6 +120,41 @@ void EntityIdVector::present(const std::string& indent)
 
 /* ****************************************************************************
 *
+* EntityIdVector::lookup - find a matching entity in the entity-vector
+*/
+EntityId* EntityIdVector::lookup(const std::string& id, const std::string& type, const std::string& isPattern)
+{
+  //
+  // isPattern:  "false" or "" is the same
+  //
+  std::string isPatternFromParam = isPattern;
+  if (isPatternFromParam == "")
+  {
+    isPatternFromParam = "false";
+  }
+
+  for (unsigned int ix = 0; ix < vec.size(); ++ix)
+  {
+    std::string isPatternFromVec = vec[ix]->isPattern;
+
+    if (isPatternFromVec == "")
+    {
+      isPatternFromVec = "false";
+    }
+
+    if ((vec[ix]->id == id) && (vec[ix]->type == type) && (isPatternFromVec == isPatternFromParam))
```

Comment:
Good point.
The way it is used (after receiving the output from mongoQueryContext) this function *should* do all that is needed for the purpose. However, with the generic name **lookup**, it should probably be 100% generic ...
I'll try to make it generic (with your help).